### PR TITLE
Refactor Step interface and introduce TransactionalStep - Issue #503

### PR DESCRIPTION
**Pull request by Marczeeee**
Issue #503

### CODE REVIEW

**Comment by davvd** [comment]
Many thanks, I'll find someone to review it before we merge

**Comment by davvd** [comment]
@krzyk please review, thanks

File: src/main/java/com/thindeck/api/Step.java [hunk 1]
```diff
@@ -37,10 +37,8 @@
  * Step in a {@link Task}.
  *
  * <p>A step is called by {@link Txn}, when it has control of a task. A
- * call is made either to {@link #exec(Context)}, {@link #commit(Context)}
- * or {@link #rollback(Context)}. This depends on the situation with
- * the transaction. This decision is made only by the transaction. The step
- * is a passive component in this sense.
+ * call is made to {@link #exec(Context)}. The step is a passive component in
+ * this sense.
```
Comment (krzyk) [inline src/main/java/com/thindeck/api/Step.java]: Remove also mention of:
> <p>Read more about our two-phase commit protocol in {@link Txn}.

File: src/main/java/com/thindeck/api/TransactionalStep.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * Transactional step in a {@link Task}.
+ *
+ * <p>A step is called by {@link Txn}, when it has control of a task. A
+ * call is made either to {@link #exec(Context)}, {@link #commit(Context)}
+ * or {@link #rollback(Context)}. This depends on the situation with
+ * the transaction. This decision is made only by the transaction. The step
+ * is a passive component in this sense.
+ *
+ * <p>A step should try to finish its execution as soon as possible, preferably
```
Comment (krzyk) [inline src/main/java/com/thindeck/api/TransactionalStep.java]: Please remove this paragraph, and add a `@see` to `Step` for more explanation

**Comment by krzyk** [comment]
@Marczeeee just few cleanup comments above

**Comment by Marczeeee** [comment]
@krzyk Updated the code according to you comments

**Comment by Marczeeee** [comment]
@krzyk Fixed it.

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by yegor256** [comment]
@rultor merge pls

**Comment by davvd** [comment]
@krzyk **15 mins** sent to your balance (ID `50843256`), many thanks!. added +15 to your rating, now it is equal to [+2929](http://www.netbout.com/b/35035?open=rating)

**Comment by davvd** [comment]
@rultor deploy pls

**Pull request merged**